File: presentation/src/main/java/biz/belcorp/consultoras/feature/home/tutorial/TutorialCardFragment.java
package biz.belcorp.consultoras.feature.home.tutorial;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import biz.belcorp.consultoras.R;
import biz.belcorp.consultoras.base.BaseFragment;
import biz.belcorp.consultoras.common.model.tutorial.TutorialModel;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * @author andres.escobar on 3/08/2017.
 */
public class TutorialCardFragment extends BaseFragment {

    @BindView(R.id.ivw_tutorial_imagen)
    ImageView ivwTutorialImagen;
    @BindView(R.id.tvw_tutorial_title_1)
    TextView tvwTutorialTitle1;
    @BindView(R.id.tvw_tutorial_title_2)
    TextView tvwTutorialTitle2;
    @BindView(R.id.tvw_tutorial_title_3)
    TextView tvwTutorialTitle3;
    @BindView(R.id.tvw_tutorial_description)
    TextView tvwTutorialDescription;

    private Unbinder unbinder;
    private TutorialModel tutorialModel;

    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_tutorial_card, container, false);
        unbinder = ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        if (getTutorialModel() == null) return;

        tvwTutorialTitle1.setText(getTutorialModel().getTitle1());
        tvwTutorialTitle2.setText(getTutorialModel().getTitle2());
        tvwTutorialTitle3.setText(getTutorialModel().getTitle3());
        tvwTutorialDescription.setText(getTutorialModel().getDescripcion());

        Glide.with(this).load(getTutorialModel().getImagen()).into(ivwTutorialImagen);
    }

    @Override
    public void onDestroyView() {
        if (unbinder != null)
            unbinder.unbind();
        super.onDestroyView();
    }

    /**********************************************************************************************/

    @Override
    public Context context() {
        return getContext();
    }

    public TutorialModel getTutorialModel() {
        return tutorialModel;
    }

    public void setTutorialModel(TutorialModel tutorialModel) {
        this.tutorialModel = tutorialModel;
    }
}

File: presentation/src/main/java/biz/belcorp/consultoras/feature/caminobrillante/feature/tutorial/TutorialCardFragment.kt
package biz.belcorp.consultoras.feature.caminobrillante.feature.tutorial


import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import biz.belcorp.consultoras.R
import biz.belcorp.consultoras.base.BaseFragment
import biz.belcorp.consultoras.common.model.tutorial.TutorialModel
import biz.belcorp.mobile.components.design.button.Button
import butterknife.ButterKnife
import butterknife.Unbinder
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.fragment_tutorial_card_camino_brillante.*


class TutorialCardFragment : BaseFragment() {

    private var unbinder: Unbinder? = null
    var tutorialModel: TutorialModel? = null
    var listener: TutorialFragment.TutorialCaminoBrillanteListener? = null

    override fun context(): Context {
        return activity?.applicationContext!!
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_tutorial_card_camino_brillante, container, false)
        unbinder = ButterKnife.bind(this, view)
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        tutorialModel?.let {
            title_tutorial_camino_brillante.text = it.title1
            tutorial_description_camino_brillante.text = it.descripcion

            Glide.with(this).load(it.imagen).into(tutorial_image_camino_brillante)
        }

        btn_omitir_camino_brillante.buttonClickListener = object : Button.OnClickListener {
            override fun onClick(view: View) {
                listener?.removeTutorial()
            }
        }
    }

    override fun onDestroyView() {
        unbinder?.unbind()
        super.onDestroyView()
    }


}
